lib/hanami/utils/inflector.rb

```diff
@@ -27,17 +27,31 @@ def add(key, value)
         # @since 0.6.0
         # @api private
         def ===(other)
-          key = other.downcase
+          key = last_alpha(other)
           @rules.key?(key) || @rules.value?(key)
         end
 
         # @since 0.6.0
         # @api private
         def apply(string)
-          key    = string.downcase
+          key = last_alpha(string)
           result = @rules[key] || @rules.rassoc(key).last
 
-          string[0] + result[1..-1]
+          prefix = if key == string.downcase
+                     string[0]
+                   else
+                     string[0..string.index(key)]
+                   end
+
+          prefix + result[1..-1]
+        end
+
+        def last_alpha(string)
```

Comment:
@jodosha I see you use `# @api private` what is that about? I wanted to put `last_alpha` under the `private` access control but I thought I'd ask about that first.